File: src/ts/Models/Genres/Genre.ts
import { default as GenreAlbum, IGenreAlbum } from '../Relations/GenreAlbum';
import { default as GenreArtist, IGenreArtist } from '../Relations/GenreArtist';

export interface IGenre {
    Id: number;
    Name: string;
    LowerName: string;
    Uri: string;
    GenreArtists: IGenreArtist[];
    GenreAlbums: IGenreAlbum[];
}

export default class Genre implements IGenre {

    public static Create(entity: IGenre): Genre {
        if (!entity)
            return null;

        const result = new Genre();
        if (entity) {
            result.Id = entity.Id || null;
            result.Name = entity.Name || null;
            result.LowerName = entity.LowerName || null;
            result.Uri = entity.Uri || null;
            result.GenreArtists = GenreArtist.CreateArray(entity.GenreArtists);
            result.GenreAlbums = GenreAlbum.CreateArray(entity.GenreAlbums);
        }

        return result;
    }

    public static CreateArray(entities: IGenre[]): Genre[] {
        const result: Genre[] = [];

        if (!entities)
            return result;

        for (let i = 0; i < entities.length; i++) {
            const entity = Genre.Create(entities[i]);
            if (entity)
                result.push(entity);
        }

        return result;
    }

    public Id: number = null;
    public Name: string = null;
    public LowerName: string = null;
    public Uri: string = null;
    public GenreArtists: GenreArtist[] = [];
    public GenreAlbums: GenreAlbum[] = [];
}

File: src/aspCore/Models/Genres/Genre.cs
using MopidyFinder.Models.Bases;
using MopidyFinder.Models.Relations;
using Newtonsoft.Json;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MopidyFinder.Models.Genres
{
    [Table("genres")]
    [JsonObject(MemberSerialization.OptIn)]
    public class Genre : IEntity
    {
        [Key]
        [JsonProperty("Id")]
        public int Id { get; set; }

        [Required]
        [JsonProperty("Name")]
        public string Name { get; set; }

        [Required]
        [JsonProperty("LowerName")]
        public string LowerName { get; set; }

        [Required]
        [JsonProperty("Uri")]
        public string Uri { get; set; }

        [JsonProperty("GenreArtists")]
        public List<GenreArtist> GenreArtists { get; set; }

        [JsonProperty("GenreAlbums")]
        public List<GenreAlbum> GenreAlbums { get; set; }
    }
}
